app/services/hyrax/characterization/file_set_description.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Hyrax
+  module Characterization
+    ##
+    # @api public
+    class FileSetDescription
+      include Hydra::Works::MimeTypes
+
+      ##
+      # @!attribute [rw] file_set
+      #   @return [Hyrax::FileSet]
+      attr_accessor :file_set
+
+      delegate :mime_type, to: :primary_file
+
+      ##
+      # @param [Hyrax::FileSet] file_set
+      # @param [Symbol] primary_file  a symbol mapping to the file_set member
+      #   used for characterization
+      def initialize(file_set:, primary_file: :original_file)
+        self.file_set = file_set
+
+        @primary_file_type_uri =
+          Hyrax::FileMetadata::Use.uri_for(use: primary_file)
```

Comment:
In other places, the file type is sometimes passed as a symbol (e.g. :original_file, :extracted_file, :thumbnail_image), and sometimes passed as a URI for custom types.  Do you think that might happen here too?  FileActor which calls Hydra::Works::AddFileToFileSet handles both symbol and URI.

Reply:
@no-reply This is an open question for me.

Reply:
i'm not sure i understand the question. this is a new interface, so callers will presumably use it as defined.

Reply:
Custom types do not have a Symbol equivalent.  So what happens here if someone wants to pass http://pcdm.org/use#remastered?

Reply:
Based on recommendations in Issue #4228 addressing inconsistencies, I would suggest that this new interface accept RDF::URI for primary_file.  And I would recommend renaming primary_file to primary_use to be consistent with the naming recommendations in #4228.